parsing/parser.mly

```diff
@@ -3162,11 +3162,13 @@ generalized_constructor_arguments:
   | COLON typevar_list DOT atomic_type %prec below_HASH
                                   { ($2,Pcstr_tuple [],Some $4) }
 ;
-
-constructor_arguments:
-  | tys = inline_separated_nonempty_llist(STAR, atomic_type)
+ constructor_arguments:
+  | atomic_type_without_open
+    %prec below_HASH
+      { Pcstr_tuple [$1] }
+  | ty = atomic_type STAR tys = inline_separated_nonempty_llist(STAR, atomic_type)
```

Comment:
Remark: we could also disallow `Foo of M.( x ) * N.( y )` to make it less surprising than `Foo of M.(x)` is a syntax error.

Reply:
Hmm, or maybe it would be possible to allow `Foo of M.(x)`. I will try that and come back to confirm that it is hell.

Reply:
I would argue that having `Foo of M.(x -> y)` allowed but not `Foo of M.( x * y)` would not be an improvement in term of the readability of the grammar.